### PR TITLE
Python script

### PR DESCRIPTION
**Pull request by taokayan**
Enhance the eosiod_run_test.py:
- allow to test with remote eosiod servers.
- If "issue" results in unexpected balance, we try to issue again(to handle the case when eosio already has some balance before the test)
- more startup parameters, allow not to kill eosiowd after the test. This is good when someone wants to run some command manually after the test.

### CODE REVIEW

File: tests/testUtils.py [hunk 1]
```diff
@@ -441,22 +441,28 @@ def createInitAccounts(self):
...
-        trans=self.pushMessage(contract, action, data, opts)
-        transId=Node.getTransId(trans[1])
-        self.waitForTransIdOnNode(transId)
-
-        expectedAmount=10000000000000
-        Utils.Print("Verify eosio issue, Expected: %d" % (expectedAmount))
-        actualAmount=self.getAccountBalance(eosio.name)
-        if expectedAmount != actualAmount:
-            Utils.errorExit("Issue verification failed. Excepted %d, actual: %d" % (expectedAmount, actualAmount))
+           Utils.errorExit("Failed to publish eosio.system.")
```
Comment (cj-oci) [inline tests/testUtils.py]: Tab space seems to be off here.

File: tests/eosiod_run_test.py [hunk 2]
```diff
@@ -65,16 +68,19 @@ def cmdError(name, code=0, exitNow=False):
...
-cluster=testUtils.Cluster(walletd=True, enableMongo=enableMongo, initaPrvtKey=initaPrvtKey, initbPrvtKey=initbPrvtKey)
-walletMgr=testUtils.WalletMgr(True)
+cluster=testUtils.Cluster(walletd=True, enableMongo=enableMongo, initaPrvtKey=initaPrvtKey, initbPrvtKey=initbPrvtKey, host=server, port=port)
+walletMgr=testUtils.WalletMgr(True, port, server)
 
 testSuccessful=False
+killEosInstances=True
+killWallet=not args.keep_wallet
 killEosInstances=not dontKill
-killWallet=not dontKill
```
Comment (cj-oci) [inline tests/eosiod_run_test.py]: Can you tie killWallet with both keep_wallet and don't-kill so users don't need to set both to leave the cluster standing? That was the original behavior.

File: tests/testUtils.py [hunk 3]
```diff
@@ -441,22 +441,28 @@ def createInitAccounts(self):
...
+            
+            expectedAmount=10000000000000+oldAmount
+            Utils.Print("Verify eosio issue, Expected: %d" % (expectedAmount))
+            actualAmount=self.getAccountBalance(eosio.name)
+            if expectedAmount == actualAmount:
+                break
+            if expectedAmount != actualAmount and oldAmount > 0:
+                Utils.errorExit("Issue verification failed. Excepted %d, actual: %d" % (expectedAmount, actualAmount))
+            oldAmount = actualAmount
+            Utils.Print("Try issue again with oldAmount: %d" % (oldAmount))
```
Comment (cj-oci) [inline tests/testUtils.py]: I feel that you need to bound this loop somehow, maybe a configurable time bound so we don't have an infinite loop.

**Comment by brianjohnson5972** [comment]
@taokayan We have had a lot of changes in this script, are you still working on this or should we close this PR?

**Comment by heifner** [comment]
@taokayan Can we get some resolution on this PR. Please update.

**Comment by cj-oci** [comment]
I recommend dropping this PR at this time. I never heard back on my review comments. At this time these changes are outdated.

**Comment by taokayan** [comment]
Sorry for my late reply. I think this can be ignored at the moment.

**Comment by heifner** [comment]
Closing this for now.

**Pull request closed without merging**